src/BLSPubkeyRegistry.sol

```diff
@@ -103,28 +117,32 @@ contract BLSPubkeyRegistry is BLSPubkeyRegistryStorage {
         for (uint i = 0; i < quorumNumbers.length; ) {
             uint8 quorumNumber = uint8(quorumNumbers[i]);
 
-            uint256 quorumApkUpdatesLength = quorumApkUpdates[quorumNumber].length;
-            if (quorumApkUpdatesLength > 0) {
-                // update nextUpdateBlockNumber of the current latest ApkUpdate
-                quorumApkUpdates[quorumNumber][quorumApkUpdatesLength - 1].nextUpdateBlockNumber = uint32(block.number);
-            }
+            // Validate quorumNumber
+            uint256 historyLength = quorumApkUpdates[quorumNumber].length;
+            require(historyLength != 0, "BLSPubkeyRegistry._processQuorumApkUpdate: quorum does not exist");
 
-            apkAfterUpdate = quorumApk[quorumNumber].plus(point);
+            // Update the last entry to point at the current block
+            // TODO - if the last entry was made in this block, update the entry instead
+            quorumApkUpdates[quorumNumber][historyLength - 1].nextUpdateBlockNumber = uint32(block.number);
 
-            //update aggregate public key for this quorum
+            // Update aggregate public key for this quorum
+            apkAfterUpdate = quorumApk[quorumNumber].plus(point);
             quorumApk[quorumNumber] = apkAfterUpdate;
-            //create new ApkUpdate to add to the mapping
-            ApkUpdate memory latestApkUpdate;
-            latestApkUpdate.apkHash = bytes24(BN254.hashG1Point(apkAfterUpdate));
-            latestApkUpdate.updateBlockNumber = uint32(block.number);
-            quorumApkUpdates[quorumNumber].push(latestApkUpdate);
+
+            // Push update to history
+            quorumApkUpdates[quorumNumber].push(ApkUpdate({
+                apkHash: bytes24(BN254.hashG1Point(apkAfterUpdate)),
+                updateBlockNumber: uint32(block.number),
+                nextUpdateBlockNumber: 0
+            }));
 
             unchecked {
                 ++i;
             }
         }
     }
 
+    // TODO - should this fail if apkUpdate.apkHash == 0? This will be the case for the first entry in each quorum
```

Comment:
good question. relatedly, should we actually push the hash of the zero point in the first update, rather than zero per se?
I think this probably _shouldn't_ revert in this case, but am not strongly opinionated

Reply:
question for @gpsanant

Reply:
so to confirm, the change is to push the zero hash for the first update, and have this method fail if `apkHash == 0`?

Should it also fail if `apkHash == ZERO_HASH`?

Reply:
so to confirm, the change is to push the zero hash for the first update, and have this method fail if `apkHash == 0`?

Should it also fail if `apkHash == ZERO_HASH`?